doc/using_poetry_and_the_python_src_layout_for_aws_lambdas.md

```diff
@@ -0,0 +1,334 @@
+
+
+1. First we export the dependencies from Poetry so that we can later use `pip install` to create the files needed for the zip image.
+    ```
+    poetry export -f requirements.txt --output requirements.txt  --without-hashes
+    ```
+1. Have Poetry build all the wheels and such
+    ```
+    poetry build
+    ```
+This will result in a bunch of files in the directory `dist` the top level of your project.
+1. Use pip to create the packages
+    ```
+    poetry run pip install -r requirements.txt --upgrade --only-binary :all: --platform manylinux2014_x86_64 --target package dist/*.whl
+    ```
+This will generate all the wheels of all the dependencies in the `package` directory, suitable for zipping into the lambda image
+
+1. Zip up the image
+    ```
+    cd package
+    mkdir -p out
+    zip -r -q out/my-lambda.zip . -x '*.pyc' out
+    cd ..
+    ```
+This will result in a file in `my_repo/services/my_lambda/package/out/my-lambda.zip` that is suitable for uploading to AWS as the Lambda Function image.
+
+### Create the Lambda function
+
+You could do this in the AWS console or use the following AWS CLI Commands (If you use the AWS Console, it will create the execution role and trust policy automatically by default).
+
+1. Create the [execution role and trust policy](https://docs.aws.amazon.com/lambda/latest/dg/gettingstarted-awscli.html#with-userapp-walkthrough-custom-events-create-iam-role)
+    ```
+    aws iam create-role --role-name my-lambda-ex --assume-role-policy-document '{"Version": "2012-10-17","Statement": [{ "Effect": "Allow", "Principal": {"Service": "lambda.amazonaws.com"}, "Action": "sts:AssumeRole"}]}'
+    ```
+    * You should get a result like the following.
+      * __Copy the arn from this output and you will use it for the next command__
+    ```json
+    {
+        "Role": {
+            "Path": "/",
+            "RoleName": "my-lambda-ex",
+            "RoleId": "AROAWUWOOBVLDKY7ZE7P3",
+            "Arn": "arn:aws:iam::1234567890123:role/my-lambda-ex",
+            "CreateDate": "2023-01-12T06:08:22+00:00",
+            "AssumeRolePolicyDocument": {
+                "Version": "2012-10-17",
+                "Statement": [
+                    {
+                        "Effect": "Allow",
+                        "Principal": {
+                            "Service": "lambda.amazonaws.com"
+                        },
+                        "Action": "sts:AssumeRole"
+                    }
+                ]
+            }
+        }
+    }
+    ```
+1. Create the Lambda function
+    * You have to set the `handler` to the 2 dot style `my_lambda.handler.handler`
+    * Change the fake account-id (`1234567890123`) in the `role arn` to your AWS account-id
+    ```shell
+    aws lambda create-function --function-name my-lambda \
+    --zip-file fileb://package/out/my-lambda.zip --handler my_lambda.handler.handler --runtime python3.9 \
+    --role arn:aws:iam::1234567890123:role/my-lambda-ex
```

Comment:
might be easier to read with a little indentation on the subsequent lines

Reply:
Fixed, though it is a shell  command which normally wouldn't have indentation, but will apply some artistic license so it looks better.